src/modules/flow/grove/grove.c

```diff
@@ -20,15 +20,21 @@
 #include <math.h>
 
 #include "sol-flow/grove.h"
-#include "sol-flow/aio.h"
 
 #include "sol-flow-internal.h"
 #include "sol-flow-static.h"
-#include "sol-i2c.h"
 #include "sol-mainloop.h"
 #include "sol-util-internal.h"
 #include "sol-vector.h"
 
+#ifdef USE_I2C
+#include "sol-i2c.h"
```

Comment:
i know, but at least it's a "section". Alternatively you can split in their own file and conditionally include them.